portfolio/src/main/webapp/script.js

```diff
@@ -91,20 +93,24 @@ function fetchComments(quantity=5) {
         }
 
         // Display image as well, if there is an image
-        let imageUrl = commentData[i].attachedImage;
-        commentContent += '<div class="comment"><div class="flex-item"><p class="body-text"><b>' + commentAuthor + '</b></p>'
-            + authorEmailContent
-            + '<p class="body-text">' + commentText + '</p></div>'; // outer <div> not closed yet
-
-        if (imageUrl == null) {
-          commentContent += '</div>'; // close outer div
-        } else {
-          commentContent += '<div class="flex-item"><a href=' + imageUrl + ' target="_blank"><img class="comment-image" src=' 
-              + imageUrl + '></a></div></div>'; // add image
-        }
+        let blobKey = commentData[i].blobKey;
+        fetchCommentImage(blobKey).then((url) => {
```

Comment:
An idea to make this nicer - you could turn fetchComments into an `async` function. Then you don't need this double nested `then()`. You don't have to do this in this pull request though.

Reply:
Yep, I'll keep this in mind and come back to it if I have time after week 5's content! 😊